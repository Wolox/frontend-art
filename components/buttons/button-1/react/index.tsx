import React from 'react';

import '../styles.scss';

function Button1() {
  return (
    <div className="column center btn-group-1">
      <button className="btn btn-1 m-bottom-2" disabled={true}>
        <div className="btn__content">cancel</div>
      </button>
      <button type="button" className="btn btn-1">
        <div className="btn__content">submit</div>
      </button>
    </div>
  );
}

export default Button1;
