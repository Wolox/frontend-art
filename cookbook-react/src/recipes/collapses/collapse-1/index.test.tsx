import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Collapse from './index';

test('render component', () => {
  const { container } = render(<Collapse />);
  expect(container).toMatchSnapshot();
});

test('collapse text', () => {
  render(<Collapse />);
  const button = screen.getByRole('button');
  userEvent.click(button);
  expect(button).toHaveClass('icon-clicked');
});
