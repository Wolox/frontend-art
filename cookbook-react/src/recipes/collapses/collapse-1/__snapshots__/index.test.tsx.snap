// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component 1`] = `
<div>
  <div
    class="collapse-container"
    id="collapse-1"
  >
    <div
      class="collapse-wrapper"
    >
      <div
        class="title-container"
      >
        <h2
          class="title"
        >
          Título / Pregunta / Ítem
        </h2>
        <button
          class="icon"
          type="button"
        />
      </div>
      <p
        class="text"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </div>
  </div>
</div>
`;
